python/ray/projects/schema.json

```diff
@@ -0,0 +1,60 @@
+{
+  "type": "object",
+  "properties": {
+    "name": {
+      "description": "The name of the project",
+      "type": "string"
+    },
+    "description": {
+      "description": "A short description of the project",
+      "type": "string"
+    },
+    "repo": {
+      "description": "The URL of the repo this project is part of",
+      "type": "string"
+    },
+    "cluster": {
+      "description": "Path to a .yaml cluster configuration file (relative to the project root)",
+      "type": "string"
+    },
+    "environment": {
+      "description": "The environment that needs to be set up to run the project",
+      "type": "object",
+      "properties": {
+        "dockerfile": {
+          "description": "Path to a Dockerfile to set up an image the project can run in (relative to the project root)",
+          "type": "string"
+        },
+        "requirements": {
+          "description": "Path to a Python requirements.txt file to set up project dependencies (relative to the project root)",
+          "type": "string"
+        },
+        "shell": {
+          "description": "A sequence of shell commands to run to set up the project environment",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "commands": {
```

Comment:
Need some example yaml in tests to see what commands means

Reply:
I scoped the commands down a little, added a description and added it to the project.yaml